lib/public/AppFramework/Db/QBMapper.php

```diff
@@ -321,6 +322,26 @@ protected function findEntities(IQueryBuilder $query): array {
 		}
 	}
 
+	/**
+	 * Runs a sql query and returns an array of entities
+	 *
+	 * @param IQueryBuilder $query
+	 * @return Generator Generator of fetched entities
+	 * @psalm-return Generator<T> Generator of fetched entities
+	 * @throws Exception
+	 * @since 30.0.0
+	 */
+	protected function yieldEntities(IQueryBuilder $query): Generator {
+		$result = $query->executeQuery();
+		try {
+			while ($row = $result->fetch()) {
```

Comment:
```suggestion
			while (($row = $result->fetch()) !== false) {
```
0, "" or '0' are valid values.

Reply:
but for entities it has to be arrays? :P

Reply:
Yep, we use `PDO::FETCH_ASSOC` as the default mode which returns `array<string,mixed>|false`

Reply:
Hum okay I missed that it was specific to entities.
Should we directly add a `$result->yieldRows()`?

How about an empty array, is that a valid row value?

I would still prefer the strict comparison to false, because this kind of code gets copy/pasted around and then we get bugs.

Reply:
> How about an empty array, is that a valid row value?

An entity without data?